cfme/tests/v2v/test_v2v_migrations.py

```diff
@@ -441,3 +299,283 @@ def test_conversion_host_tags(appliance, v2v_providers):
+    logger.info("For plan %s, migration status after completion: %s, total time elapsed: %s",
+        migration_plan.name, view.migration_plans_completed_list.get_vm_count_in_plan(
+            migration_plan.name), view.migration_plans_completed_list.get_clock(
+            migration_plan.name))
+    assert view.migration_plans_completed_list.is_plan_succeeded(migration_plan.name)
+
+
+@pytest.mark.parametrize('form_data_multiple_vm_obj_single_datastore', [['nfs', 'nfs',
+    [ubuntu16_template, rhel7_minimal]]],
+    indirect=True)
+def test_dual_vm_migration_cancel_migration(request, appliance, v2v_providers, host_creds,
+                                        conversion_tags,
+                                        form_data_multiple_vm_obj_single_datastore, soft_assert):
+    # This test will make use of migration request details page to track status of migration
+    infrastructure_mapping_collection = appliance.collections.v2v_mappings
+    mapping = infrastructure_mapping_collection.create(
+        form_data_multiple_vm_obj_single_datastore[0])
+
+    @request.addfinalizer
+    def _cleanup():
+        infrastructure_mapping_collection.delete(mapping)
+
+    migration_plan_collection = appliance.collections.v2v_plans
+
+    migration_plan = migration_plan_collection.create(
+        name="plan_{}".format(fauxfactory.gen_alphanumeric()), description="desc_{}"
+        .format(fauxfactory.gen_alphanumeric()), infra_map=mapping.name,
+        vm_list=form_data_multiple_vm_obj_single_datastore[1], start_migration=True)
+    # as migration is started, try to track progress using migration plan request details page
+    view = appliance.browser.create_view(navigator.get_class(migration_plan_collection, 'All').VIEW)
+    wait_for(func=view.progress_card.is_plan_started, func_args=[migration_plan.name],
+        message="migration plan is starting, be patient please", delay=5, num_sec=150,
+        handle_exception=True)
+    view.progress_card.select_plan(migration_plan.name)
+    view = appliance.browser.create_view(navigator.get_class(migration_plan_collection,
+                                                             'Details').VIEW)
+    view.wait_displayed()
+    request_details_list = view.migration_request_details_list
+    vms = request_details_list.read()
+
+    def _get_plan_status_and_cancel():
+        migration_plan_in_progress_tracker = []
+        for vm in vms:
+            clock_reading1 = request_details_list.get_clock(vm)
+            time.sleep(1)  # wait 1 sec to see if clock is ticking
+            logger.info("For vm %s, current message is %s", vm,
+                request_details_list.get_message_text(vm))
+            current_progress_text = request_details_list.get_progress_description(vm)
+            current_data_transferred, total_data_to_transfer = map(float, re.findall("\d+\.\d+",
+                current_progress_text))
+            if current_data_transferred > (total_data_to_transfer / 2):
```

Comment:
Can we just wait until its a 3rd done? or a 5th? 10th? Thoughts on what's reasonable here, I'm thinking 10% might really be sufficient for a process cancel test, at least starting. If its an area we want to automate for more coverage, we can start to push the limits of canceling immediately, and at 99%, etc.

Reply:
This is something I can improve on in a different PR. Definitely I'd like to try cancellation at different phases. But initially just checking functionality itself is good enough. I'll add a `#TODO` ? Will submit a PR in early next week .